lib/ansible/module_utils/facts/packages.py

```diff
@@ -0,0 +1,82 @@
+# (c) 2018, Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from abc import ABCMeta, abstractmethod
+
+from ansible.module_utils.six import with_metaclass
+from ansible.module_utils.basic import get_all_subclasses
+from ansible.module_utils.common.process import get_bin_path
+
+
+def get_all_pkg_managers():
+
+    return dict([(obj.__name__.lower(), obj) for obj in get_all_subclasses(PkgMgr) if obj not in(CLIMgr, LibMgr)])
+
+
+class PkgMgr(with_metaclass(ABCMeta, object)):
+
+    @abstractmethod
+    def is_available(self):
+        ''' This method is supposed to return True/False if the package manager is currently installed/usable'''
```

Comment:
Could you please follow PEP 257 for docstrings?

Reply:
done